### PR TITLE
Modernize comments app

### PR DESCRIPTION
**Pull request by CarlSchwan**
* Use php 7.4 typed properties
* More typed methods (and replace existing php doc in some cases)
* query -> get for Container

## Test plan

1. Unit tests are working locally
2. I can post and edit comments in the sidebar

Signed-off-by: Carl Schwan <carl@carlschwan.eu>

### CODE REVIEW

File: apps/comments/lib/Collaboration/CommentersSorter.php [hunk 1]
```diff
@@ -102,7 +96,7 @@ protected function retrieveCommentsInformation($type, $id) {
 		return $actors;
 	}
 
-	protected function compare(array $a, array $b, array $commenters) {
+	protected function compare(array $a, array $b, array $commenters): int {
```
Comment (come-nc) [inline apps/comments/lib/Collaboration/CommentersSorter.php:99]: Maybe replace `return $valueB - $valueA;` with `return $valueB <=> $valueA;`

File: apps/comments/lib/Controller/Notifications.php [hunk 2]
```diff
@@ -43,31 +43,17 @@
...
 	 * @param string $appName
-	 * @param IRequest $request
-	 * @param ICommentsManager $commentsManager
-	 * @param IRootFolder $rootFolder
-	 * @param IURLGenerator $urlGenerator
-	 * @param IManager $notificationManager
-	 * @param IUserSession $userSession
 	 */
 	public function __construct(
 		$appName,
```
Comment (come-nc) [inline apps/comments/lib/Controller/Notifications.php:59]: Any chance of strong typing $appName as string?
```suggestion
		string $appName,
```

File: apps/comments/lib/Activity/Provider.php [hunk 3]
```diff
@@ -37,34 +37,15 @@
...
 
-	/** @var IFactory */
-	protected $languageFactory;
+	protected IFactory $languageFactory;
+	protected IL10N $l;
+	protected IUrlGenerator $url;
+	protected ICommentsManager $commentsManager;
+	protected IUserManager $userManager;
+	protected IManager $activityManager;
+	/** @param string[] $displayNames */
```
Comment (come-nc) [inline apps/comments/lib/Activity/Provider.php:46]:
```suggestion
	/** @param string[] */
```

Comment (come-nc) [reply apps/comments/lib/Activity/Provider.php:46]: Also @param or @var?

**Review by come-nc — APPROVED** [review]

**Review by skjnldsv — APPROVED** [review]
Looks amazing!!!

**Pull request merged**